.github/workflows/doc_builder.yml

```diff
@@ -0,0 +1,18 @@
+name: Publish docs via GitHub Pages
+on:
+  push:
+    branches:
+      - mainline
```

Comment:
We shouldn't update our docs for every push to mainline, because we update docs for unreleased features.

Can we instead trigger this action on `vX.Y.Z` tags?